cupy/_core/core.pyx

```diff
@@ -1799,6 +1799,11 @@ cdef class ndarray:
                                  bint update_f_contiguity):
         if shape.size() != strides.size():
             raise ValueError('len(shape) != len(strides)')
+        if shape.size() > _carray.MAX_NDIM:
+            msg = f'The input array dimension ({shape.size()}) is '
+            msg += 'beyond CuPy\'s current limitation ('
+            msg += f'{_carray.MAX_NDIM})'
+            raise ValueError(msg)
```

Comment:
How about using the same error message as NumPy? 😄 

```
>>> numpy.empty((1,)*34)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: maximum supported dimension for an ndarray is 32, found 34
```

Reply:
Fixed in f534a49.